velox/functions/prestosql/tests/JsonCastTest.cpp

```diff
@@ -1581,4 +1582,24 @@ TEST_F(JsonCastTest, castFromJsonWithEscaping) {
   auto resultVarchar =
       evaluate("cast(json_parse(c0) as varchar)", makeRowVector({svData}));
   test::assertEqualVectors(expectedVarchar, resultVarchar);
+
+  // Create a large vector to ensure vectors string buffer has its capacity
+  // computed correctly.
+  {
+    auto largeVector = makeFlatVector<StringView>(1000, [](auto row) {
+      return R"({"someKey": "some large enough string 😀"})"_sv;
+    });
+    auto resultLarge = evaluate(
+        "cast(json_parse(c0) as map(varchar, json))",
+        makeRowVector({largeVector}));
+    auto resultMap = resultLarge->as<MapVector>();
```

Comment:
nit: this shadows another variable resultMap defined in the test (lint warning)

Reply:
Yes but I like this name - i will remove the auto..